graylog2-server/src/main/java/org/graylog2/rest/resources/system/GrokResource.java

```diff
@@ -96,7 +97,9 @@ public Response createPattern(@ApiParam(name = "pattern", required = true)
     @PUT
     @Timed
     @ApiOperation("Add a list of new patterns")
-    public Response bulkUpdatePatterns(@ApiParam(name = "patterns", required = true) @NotNull GrokPatternList patternList) throws ValidationException {
+    public Response bulkUpdatePatterns(@ApiParam(name = "patterns", required = true) @NotNull GrokPatternList patternList,
+                                       @ApiParam(name = "replace", value = "Replace all patterns with the new ones.")
+                                       @QueryParam("replace") boolean replace) throws ValidationException {
```

Comment:
Maybe add a default value for `replace` (just to be sure) with [`@DefaultValue`](https://jersey.java.net/apidocs/2.16/jersey/javax/ws/rs/DefaultValue.html).